src/Compilers/CSharp/Test/Semantic/Semantics/BindingTests.cs

```diff
@@ -1999,7 +1999,11 @@ unsafe static void M(int[] arg)
             CreateCompilationWithMscorlib(source, options: TestOptions.UnsafeReleaseDll).VerifyDiagnostics(
                 // (7,32): error CS1023: Embedded statement cannot be a declaration or labeled statement
                 //         fixed (int* ptr = arg) object o = null;
-                Diagnostic(ErrorCode.ERR_BadEmbeddedStmt, "object o = null;"));
+                Diagnostic(ErrorCode.ERR_BadEmbeddedStmt, "object o = null;").WithLocation(7, 32),
+                // (7,39): warning CS0219: The variable 'o' is assigned but its value is never used
+                //         fixed (int* ptr = arg) object o = null;
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "o").WithArguments("o").WithLocation(7, 39)
+                );
```

Comment:
I was puzzled at first too. The change ensures that local declaration statement used as an embedded statement always gets surrounded with a BoundBlock that represents a scope for the locals declared by the statement. Before, we had inconsistency, LocalBinderFactory would create a binder for the scope, but Bonder.BindPossibleEmbeddedStatement wouldn't create a block for that scope. So the bound tree was inconsistent - scope owner for the locals was not present in the tree. The change fixes this inconsistency. As a result, flow analysis detects declarations and properly tracks their scope.